### PR TITLE
Azure auth

### PR DESCRIPTION
**Pull request by not-night-but**
Fix [azure sso auth](https://github.com/beekeeper-studio/ultimate/issues/72)

First pass at this, definitely still some stuff to work on, but I'm wanting some feedback.

# TODO
- [x] Figure out what happens if perms expire/are revoked with an active connection. May need to add reauthentication to the error handler on the pool.
- [x] Fix tests
- [ ] Make sure that it is ult only
- [x] Add new field to cloud
- [ ] Make sure that the tokenCache is regenerated for every new device the connection is on
- [x] Make the official azure application

# NOTES
- I just have the clientid hardcoded in the azureAuthService (it is inserted as plaintext in the url, so not exactly a secret?)
- To actually test this, you'll have to comment out the ultimate checking in SqlServerForm
- I put the token cache in a new table to avoid updating the connection record while it's active (as we don't update that object in memory). This also makes it easier to grab as we have the id on the connection so its trivial, but trying to grab the connection record with the information available at that point is reasonably difficult (no id, no sense of whether its a used connection or saved connection, etc). This also prevents polluting that object, as the cache is one heck of a string

----

| :rocket: This description was created by [Ellipsis](https://www.ellipsis.dev) for commit 656494eb66941cce40837f7226bf53f9ee25cd9b  | 
|--------|

### Summary:
This PR integrates Azure SSO authentication for SQL Server connections in Beekeeper Studio, including UI updates, token caching, and handling of token refresh scenarios.

**Key points**:
- Integrated Azure SSO authentication for SQL Server connections.
- Added new columns for token caching in database models.
- Updated connection handling to support Azure authentication.
- Handled token refresh and reauthentication scenarios.
- Updated UI components to allow Azure SSO configuration.
- Added new migrations for database changes.

----
Generated with :heart: by [ellipsis.dev](https://www.ellipsis.dev)

### CODE REVIEW

**Comment by rathboma** [comment]
@ellipsis-dev code review please

**Review by rathboma — CHANGES_REQUESTED** [review]
Looks awesome! Super clean.

had a a about foreign keys, plus a couple of nits, but otherwise good

File: apps/studio/src/components/ConnectionInterface.vue [hunk 1]
```diff
@@ -421,6 +426,15 @@ export default Vue.extend({
         if (!this.config.name) {
           throw new Error("Name is required")
         }
+        // create token cache for azure auth
+        if (this.config.azureAuthOptions.azureAuthEnabled && !this.config.azureAuthOptions.authId) {
+          console.log('creating cache')
+          let cache = new TokenCache();
+          cache = await cache.save();
+          console.log('saved cache: ', cache)
+          this.config.azureAuthOptions.authId = cache.id;
```
[the remark this answers is not in the document]
Comment (not-night-but) [reply apps/studio/src/components/ConnectionInterface.vue:435]: @rathboma I've moved it, but sqlite doesn't support adding relations after the table has been created. If we really want to do it I could write a script to recreate the table with that fk, up to you

[the remark this answers is not in the document]
Comment (rathboma) [reply apps/studio/src/components/ConnectionInterface.vue]: @not-night-but just change your migration to do it the way we want, then kill your ~/.config/beekeeper-studio/app-dev.db to test it

**Comment by rathboma** [comment]
@not-night-but if you fix merge conflicts, and then fix the migration, we can merge this and release it!

File: apps/studio/src/components/ConnectionInterface.vue (hunk 1 above)
[the remark this answers is not in the document]
Comment (not-night-but) [reply apps/studio/src/components/ConnectionInterface.vue]: @rathboma sorry I'm confused. Don't you want the FK on saved/used connections?

**Review by rathboma — APPROVED** [review]
Let's gooooo

**Comment by vamsitp** [comment]
Any plan to provide Azure AD (SSO) support for Azure PGSQL too?

**Comment by rathboma** [comment]
I didn't even know that was a thing, but yes we can do that :-)

**Comment by vamsitp** [comment]
https://techcommunity.microsoft.com/t5/azure-database-for-postgresql/azure-active-directory-authentication-in-azure-postgresql/ba-p/3723066

**Pull request merged**